src/core/model/types/numeric_type.h

```diff
@@ -29,8 +35,89 @@ class INumericType : public IMetrizableType {
 
     [[nodiscard]] virtual std::byte const* Min() const = 0;
     [[nodiscard]] virtual std::byte const* Max() const = 0;
+
+private:
+    inline void IntCastTo_(std::byte* value, TypeId to_type) const;
+    inline void DoubleCastTo_(std::byte* value, TypeId to_type) const;
```

Comment:
These methods seems redundant. It should be possible to make `void CastTo(std::byte* value, TypeId to_type)` virtual and perform conversions in `IntType::CastTo` and `DoubleType::CastTo` overrides correspondignly